### DISCUSSION TITLE
Doesn't respond to music on my Fire Stick

### DISCUSSION THREAD

**Discussion by rhodesengr** [projectM-visualizer/projectm#850] [Q&A] [answered]
I just loaded ProjectM on my Fire stick to visualize music streaming on Pandora. The app runs after i start Pandora and displays screens but it is not actually responding to the music in any obvious way. I see numerous reviews on Amazon saying the same thing. Is there a way to fix this? It's a paid app so it should work better.

**Comment by rhodesengr** [comment, marked as answer]
I solved it with this command 
./adb shell pm grant com.psperl.projectMTV android.permission.RECORD_AUDIO

But now I want to get my windows version working. It also does not respond to music at all.

**Reply by kblaschke** [reply]
For Windows, you can use [this development release](https://github.com/kblaschke/frontend-sdl2/releases/tag/2.0-windows-pre3). The Windows Store versions is super old and has all the limitations imposed on store apps (restricted access to audio streams, only using OpenGL ES). The dev build is running as a native application and can use desktop OpenGL and has access to all audio devices (even supports hotplugging).

**Reply by rhodesengr** [reply]
OK Thanks. I will try this version today. I have to correct what I said before a little. I found two versions for Windows. One was on Steam and the other was on Microsoft Store. The version from Microsoft actually seems to work OK. The one on Steam does not seem to respond to the audio at all. They have somewhat different hotkey controls also. The Microsoft one does not have the cntl I function that changes the audio source so I don't know if it is using the mic or internal stream but it does seem to respond. Sometimes it is a little hard  to tell. Most of the presets keep changing even when there is no music.

**Reply by rhodesengr** [reply]
the development version doesn't seem to run on my laptop. I get a screen that opens that stays black. If I hit any of the control keys, I get a "not responding" at the top of the window. I did install the C++ and rebooted.

**Reply by kblaschke** [reply]
Laptops generally have issues on Windows with software using OpenGL, because the GPU selection is done by the OS and might select an implementation (even a software one) that doesn't really work with the application. Sadly, OpenGL has no way of choosing a specific GPU or GL implementation, you just have to use what you get.
So from my side, it's hard to tell without knowing what exactly goes wrong, but my guess is some OpenGL driver issue on the laptop. If it was a runtime problem, the main window wouldn't even appear. In such a case, I'd recommend using [BeatDrop](https://github.com/OfficialIncubo/BeatDrop-Music-Visualizer), as it uses DirectX 9 and the original Milkdrop codebase. It also has the same issues with audio capturing though, not sure if Incubo already added the WASAPI implementation from the projectM standalone app linked above, see issue OfficialIncubo/BeatDrop-Music-Visualizer#14 for that one.

**Reply by OfficialIncubo** [reply]
Nah. @mvsoft74 made BeatDrop integrating with WASAPI (loopback-capture code) many years ago. It wasn't me. 😅
...and I didn't add WASAPI implementation from projectM because I get too many errors.

**Reply by rhodesengr** [reply]
Beatdrop doesn't run on my machine. I installed the directX and ran the Beatdrop installer. When I try to run it, I see nothing. No screen opens up.

**Reply by OfficialIncubo** [reply]
Then what operating system are you using? You can try the portable version of it!

**Reply by rhodesengr** [reply]
Its Windows 11. Where can I get the portable version?
Edit: I found the portable version but it also doesn't work on this laptop. No screen opens up. Here are my system details
Device name	LAPTOP-OCBLNMC3
Processor	Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz   2.59 GHz
Installed RAM	16.0 GB (15.9 GB usable)
Device ID	7519D9A0-665E-472B-8FA3-DCE6BBFCE768
Product ID	00325-81346-73275-AAOEM
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

Edition	Windows 11 Home
Version	23H2
Installed on	‎10/‎26/‎2022
OS build	22631.4460
Experience	Windows Feature Experience Pack 1000.22700.1047.0

**Reply by kblaschke** [reply]
If neither projectM not BeatDrop work, I guess there's a general issue with that laptop or Windows installation. It's hardly the case that two completely different 3D APIs fail in basically the exact same way.

The built-in GPU only has 64 MB of VRAM available, which clearly isn't enough for any kind of 3D application, so to me it looks like that GPU is just capable of displaying 2D graphics and possibly some simple 3D graphics in very low resolution. Just the two framebuffers required for rendering would take more than the available GPU memory on anything above 720p.

**Reply by rhodesengr** [reply]
I don't know what the differences are but one version of projectM does work on that laptop, the one from Microsoft Store. Another similar program called Kauna partially works. The graphics work fine but the sound only works from the microphone. Not sure if that tells you anything.

**Reply by kblaschke** [reply]
The store version uses OpenGL ES 2, which is the OpenGL version tailored specifically to low-power embedded devices which don't have a powerful GPU.

Since the Windows version we currently provide uses desktop GL (version 3.3), this might not be supported by the laptop GPU or drivers. It's possible to compile a recent projectM with OpenGL ES, but we currently don't have pre-built binaries for Windows that use it. We may provide them later on GitHub, but we no longer have access to the Window Store account, so this version there will sadly never receive an update.
